src/org/zaproxy/zap/extension/api/API.java

```diff
@@ -289,13 +290,17 @@ public boolean handleApiRequest (HttpRequestHeader requestHeader, HttpInputStrea
 
 					ApiResponse res;
 					switch (reqType) {
-					case action:	
-						// TODO Handle POST requests - need to read these in and then parse params from POST body
-						/*
-						if (getOptionsParamApi().isPostActions()) {
-							throw new ApiException(ApiException.Type.DISABLED);
+					case action:
+						if (requestHeader.getMethod().equalsIgnoreCase(HttpRequestHeader.POST)) {
+							if(requestHeader.getHeader(HttpHeader.CONTENT_TYPE)
+									.equals(HttpHeader.FORM_URLENCODED_CONTENT_TYPE)) {
+								String body = httpIn.readRequestBody(requestHeader).toString();
+								params = getParams(body);
+							} else {
+								throw new ApiException(ApiException.Type.DISABLED);
```

Comment:
It should use another type (`UNSUPPORTED_CONTENT_TYPE`?), it might be a little confusing using the one that indicates the API is disabled.